packages/flutter/lib/src/gestures/binding.dart

```diff
@@ -130,12 +130,21 @@ mixin GestureBinding on BindingBase implements HitTestable, HitTestDispatcher, H
     } else if (event is PointerUpEvent || event is PointerCancelEvent) {
       result = _hitTests.remove(event.pointer);
     } else if (event.down) {
+      // In addition to PointerDownEvents, this block also handles
+      // PointerMoveEvents. Because PointerMoveEvents should be dispatched to
+      // the same place that their initial PointerDownEvent was, we want to
+      // re-use the path we found when the pointer went down, rather than do hit
+      // detection each time we get a PointerMoveEvent.
       result = _hitTests[event.pointer];
-    } else {
-      return; // We currently ignore add, remove, and hover move events.
     }
-    if (result != null)
+    assert(() {
+      if (debugPrintMouseHoverEvents && event is PointerHoverEvent)
+        debugPrint('$event');
+      return true;
+    }());
+    if (result != null || event is PointerHoverEvent || event is PointerAddedEvent || event is PointerRemovedEvent) {
       dispatchEvent(event, result);
+    }
```

Comment:
this seems like a major change? don't we want to be dispatching other events anymore?

Reply:
It's not a major change: all events that have a hit test result will still be processed as before. This is saying "dispatch the event if there is hit test information [as before], or if there is a PointerHoverEvent, PointerAddedEvent or PointerRemovedEvent that doesn't have hit test information".  I have broadened the conditional this way because unlike other events, these three events are meant to be dispatched even when they don't have hit test information. They weren't being handled before.

Reply:
Oh, I see. `!= null` covers the previously covered cases. Got it.